patterns/hidden-comments.php

```diff
@@ -22,7 +22,7 @@
 				<!-- wp:group -->
 				<div class="wp-block-group">
 					<!-- wp:comment-author-name /-->
-					<!-- wp:comment-date {"format":"F j, Y \\a\\t g:i a"} /-->
+					<!-- wp:comment-date {"format":"F j, Y - g:i a"} /-->
```

Comment:
we could also wrap the whole thing in a translation function and let each translator decide?

Reply:
I was unsure on what translation function to use since I'm not familiarized with wordpress i18n, so I choosed keeping it simple initially (in my first ever pull request btw).

But I guess it should be as simple as doing:

`<!-- wp:comment-date {"format":"F j, Y <?php echo esc_html__( 'at', 'twentytwentyfour' ); ?> g:i a"} /-->`

or

`<!-- wp:comment-date {"format":"F j, Y <?php esc_html_e( 'at', 'twentytwentyfour' ); ?> g:i a"} /-->`

Reply:
if you check the [README's guidelines](twentytwentyfour/home) you see the options you have. esc_html_x is probably the best option here since it allows you to add context for the translators (check how other strings have done this over the theme). I would include the whole date format string, because for example, the comma after the day may not be something every language wants to include.

Reply:
Will it be properly rendered then if it's in a translation function?

Reply:
good question, I don't know what format is expecting, we could look into that, usually themes don't bother adding a custom format at all. Also it might as well be that GB already takes care of translating the format and we shouldn't touch this at all, I'll go check

Reply:
No I wouldn't want to put the entire date string into the translation function, as there are separate settings for date and time format in WordPress, and I'm worried they won't be taken into consideration when you hardcode it into a translation string.

Reply:
I just spotted some more issues with the dates comparing the post patterns on the home page, post meta on single and comment date.

Reply:
Looks like it was a custom format instead of choosing one of the presets:

<img width="272" alt="Screenshot 2023-09-22 at 12 36 54" src="https://github.com/WordPress/twentytwentyfour/assets/3593343/a02b51cb-39e4-4eb6-b6de-a5b0145aa0b6">

To be honest, given that it's the comment form I'd remove the format attr altogether and let the block do the work for us, that's what's been done on previous themes

Reply:
If the format is set to `null`, it will fall back to the date and time format settings found in the general settings of WordPress. We should go for this one instead of forcing certain date and time formats in the theme itself, as they vary between countries.

Reply:
I've read it, and I still was unsure haha.

But I see that in the same file on line 13 was done like this:
`<h2 class="screen-reader-text"><?php esc_html_e( 'Comments', 'twentytwentyfour' ); ?></h2>`

So that's another thing to consider, for consistence.

I tried doing it like in my comment above and it seems to work fine.

Again I have doubts that including the whole format string would make sense as a translation. I mean, a translation should include a format decision ?

About the format itself, what I can add is that often in the US they often use mm/dd/yyyy while in europe is more common to use dd/mm/yyyy so yeah, the comma and the whole format... would depend on the culture of the user.

Reply:
Just remove the format, don't add any translation stuff. This should be handled completely by WP itself. If the user wants a custom format, they can go in the editor and change it (see screenshot above by Maggie).

Reply:
Oh, I see. You both were faster than me commenting.

Would you keep the format property:
`<!-- wp:comment-date {"format":""} /-->`

or we would just do:
` <!-- wp:comment-date /-->`

Completely removing it also falls back to general settings, so I don't see any reason to keep the `{"format":""}` part.

Reply:
If it falls back to the default without the format attribute, I'm fine leaving it out.